### PR TITLE
add estimated gas logic

### PR DESCRIPTION
**Pull request by mikedoan**
add estimated gas logic

### CODE REVIEW

**Comment by LeoHChen** [comment]
failed on both CI and Jenkins.

**Comment by mikedoan** [comment]
Jenkins, test this please.

File: internal/hmyapi/blockchain.go [hunk 1]
```diff
@@ -197,3 +193,57 @@ func doCall(ctx context.Context, b Backend, args CallArgs, blockNr rpc.BlockNumb
 	}
 	return res, gas, failed, err
 }
+
+// doEstimateGas calculates estimated gas to process the transaction.
+func doEstimateGas(ctx context.Context, b Backend, args CallArgs, blockNr rpc.BlockNumber) (hexutil.Uint64, error) {
```
[the remark this answers is not in the document]
Comment (denniswon) [reply internal/hmyapi/blockchain.go]: I see. With different args, I guess the code is a little bit different. but no logic difference.

File: internal/hmyapi/blockchain.go [hunk 2]
```diff
@@ -102,11 +102,11 @@ func (s *PublicBlockChainAPI) BlockNumber() hexutil.Uint64 {
 // Call executes the given transaction on the state for the given block number.
 // It doesn't make and changes in the state/blockchain and is useful to execute and retrieve values.
 func (s *PublicBlockChainAPI) Call(ctx context.Context, args CallArgs, blockNr rpc.BlockNumber) (hexutil.Bytes, error) {
-	result, _, _, err := doCall(ctx, s.b, args, blockNr, vm.Config{}, 5*time.Second, s.b.RPCGasCap())
```
Comment (denniswon) [inline internal/hmyapi/blockchain.go]: why is this removed?

**Comment by LeoHChen** [comment]
what's the status of this one? I think we need it for token transfer.

**Comment by fxfactorial** [comment]
@mikedoan can we close this?
